gmso/abc/abstract_site.py

```diff
@@ -24,6 +24,8 @@ def default_position():
 class Site(GMSOBase):
     __iterable_attributes__: ClassVar[set] = {
         "label",
+        "group",
```

Comment:
As per discussion with @daico007 and @bc118, this group label will be just a site attribute that lets users write out the indices of that group to something like a gromacs .ndx file. However, this isn't necessary for all writers or all uses cases, so maybe instead of having it be a default attachment to every site and having to lug that around for every workflow, we could come up with a better way to store that data on a need by need basis, like something that is an optional attribute.